discovery-provider/src/tasks/entity_manager/entity_manager.py

```diff
@@ -150,11 +154,21 @@ def entity_manager_update(
                     elif (
                         params.action in create_actions and ENABLE_DEVELOPMENT_FEATURES
                     ):
-                        create_social_record(params)
+                        if should_check_entity_access(
```

Comment:
would prefer if we move this logic to `validate_social_feature` so this is more abstract and doesn't need to know about premium access